### ISSUE TITLE
Make Notification widget accessible

### ISSUE THREAD

**Issue by veselints** [telerik/kendo-ui-core#5406]
### Bug report
Currently, the Notification widget is not accessible, as the readers do not announce its contents and it cannot be focused.

### Reproduction of the problem
Run the Notification Index demo: https://demos.telerik.com/kendo-ui/notification/index. Try to navigate to the Notification while it is open. Also, the screen readers do not announce the Notification content upon open.

### Expected/desired behavior
Here are what needs to be done:

- [x] Assign `role="alert"` to the Notification container. That will force the readers to announce the Notification contents upon open;
Note that [role="alert"](https://www.w3.org/TR/wai-aria/#alert) has the following requirements related to focus:
> "Since alerts are **not required to receive focus**, content authors **SHOULD NOT require users to close an alert**. If the operating system allows, the user agent SHOULD fire a system alert event through the accessibility API when the WAI-ARIA alert is created. If an alert requires focus to close the alert, then content authors SHOULD use alertdialog instead."

Thus, with `role="alert"` the Notification will not be focusable and its close button should not be focusable and required to close the Notification. The following points should not be implemented:

~~- [ ] Allow the notification to be focused;~~
~~- [ ] Implement Notification close logic upon keyboard interaction, while the Notification is focused;~~
~~- [x] Make Notification close button accessible: set its `role="button"` and make it focusable;~~
~~- [ ] Implement Notification close logic upon pressing Enter while the close button is focused;~~
- [x] Assign `title` to the Notification element and expose the respective configuration property in the widget;
- [x] Add `aria-live="polite"` to the Notification element;

### Environment

* **Kendo UI version:** 2019.3.1023
* **Browser:** [all]

**Comment by veselints** [comment]
Also reported in Ticket ID: 1501975

**Comment by veselints** [comment]
I would suggest to follow the WAI-ARIA specification and assign role `alert` to the widget:
https://www.w3.org/TR/wai-aria/#alert

The specifications states "Since alerts are not required to receive focus, content authors SHOULD NOT require users to close an alert.". That is the default behavior of the Kendo Notification - it is being automatically closed after some time, which complies perfectly with the WAI-ARIA specification. 

We will include a note in our documentation that a Notification with [`autoHide`](https://docs.telerik.com/kendo-ui/api/javascript/ui/notification/configuration/autohideafter) set to 0 (disabled `autoHide`) would not expose a focusable close button. That follows the above WAI-ARIA spec as "authors SHOULD NOT require users to close an alert".

**Comment by veselints** [comment]
The other suggested alternative is to use `aria-live="polite"` and skip the `role="alert"`. That would result in a less assertive object. Its usability, however, is under question. It is not clear that an element that has `aria-live="polite"` could be dynamically added to the DOM (with is the case with the Notification) and whether the user agents and the screen readers are expected to announce its content on that point. Here are the very few discussion I have found on that matter so far:
https://stackoverflow.com/questions/17977690/dynamically-added-elements-for-screen-reader-aria-live-vs-role-alert
https://stackoverflow.com/questions/10637231/can-the-aria-live-attribute-of-a-given-element-itself-be-set-dynamically-with-ja

Taking into account the uncertainty of the above, I would suggest using `role="alert"` with `aria-live` property explicitly set to `popite`, which would diminish the assertiveness of the widget.

**Comment by Dimitar-Goshev** [comment]
- [ ] To be with role alert, Notification and its close button should not be focusable
- [ ] Add the title property to TypeScript
